src/intents.cs

```diff
@@ -8528,5 +8702,299 @@ interface INVisualCodeTypeResolutionResult {
+		INMediaItemType Type { get; }
+
+		[NullAllowed, Export ("artwork", ArgumentSemantic.Copy)]
+		INImage Artwork { get; }
+	}
+
+	[Watch (5,0), NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface INObject : INSpeakable, NSCopying, NSSecureCoding {
+
+		[Export ("initWithIdentifier:displayString:pronunciationHint:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] string identifier, string displayString, [NullAllowed] string pronunciationHint);
+
+		[Export ("initWithIdentifier:displayString:")]
+		IntPtr Constructor ([NullAllowed] string identifier, string displayString);
+
+		// Inlined by INSpeakable
+		//[NullAllowed, Export ("identifier", ArgumentSemantic.Strong)]
+		//string Identifier { get; }
+
+		[Export ("displayString")]
+		string DisplayString { get; }
+
+		// Inlined by INSpeakable
+		//[NullAllowed, Export ("pronunciationHint", ArgumentSemantic.Strong)]
+		//string PronunciationHint { get; }
+	}
+
+	[Watch (5,0), NoMac, iOS (12,0)]
+	[BaseType (typeof (INIntent))]
+	[DisableDefaultCtor]
+	interface INPlayMediaIntent {
+
+		[Export ("initWithMediaItems:mediaContainer:playShuffled:playbackRepeatMode:resumePlayback:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] INMediaItem [] mediaItems, [NullAllowed] INMediaItem mediaContainer, [NullAllowed, BindAs (typeof (bool?))] NSNumber playShuffled, INPlaybackRepeatMode playbackRepeatMode, [NullAllowed, BindAs (typeof (bool?))] NSNumber resumePlayback);
+
+		[NullAllowed, Export ("mediaItems", ArgumentSemantic.Copy)]
+		INMediaItem [] MediaItems { get; }
+
+		[NullAllowed, Export ("mediaContainer", ArgumentSemantic.Copy)]
+		INMediaItem MediaContainer { get; }
+
+		[BindAs (typeof (bool?))]
+		[NullAllowed, Export ("playShuffled", ArgumentSemantic.Copy)]
+		NSNumber PlayShuffled { get; }
+
+		[Export ("playbackRepeatMode", ArgumentSemantic.Assign)]
+		INPlaybackRepeatMode PlaybackRepeatMode { get; }
+
+		[BindAs (typeof (bool?))]
+		[NullAllowed, Export ("resumePlayback", ArgumentSemantic.Copy)]
+		NSNumber ResumePlayback { get; }
+	}
+
+	[Watch (5,0), NoMac, iOS (12,0)]
+	[Protocol]
+	interface INPlayMediaIntentHandling {
+
+		[Abstract]
+		[Export ("handlePlayMedia:completion:")]
+		void HandlePlayMedia (INPlayMediaIntent intent, Action<INPlayMediaIntentResponse> completion);
+
+		[Export ("confirmPlayMedia:completion:")]
+		void Confirm (INPlayMediaIntent intent, Action<INPlayMediaIntentResponse> completion);
+	}
+
+	[Watch (5,0), NoMac, iOS (12,0)]
+	[BaseType (typeof (INIntentResponse))]
+	[DisableDefaultCtor]
+	interface INPlayMediaIntentResponse {
+
+		[Export ("initWithCode:userActivity:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (INPlayMediaIntentResponseCode code, [NullAllowed] NSUserActivity userActivity);
+
+		[Export ("code")]
+		INPlayMediaIntentResponseCode Code { get; }
+
+		[NullAllowed, Export ("nowPlayingInfo", ArgumentSemantic.Copy)]
+		NSDictionary WeakNowPlayingInfo { get; set; }
```

Comment:
Could we add a strong dictionary here?

Reply:
Yes, it's on my TODOs to figure out the keys that are available here, the comment is a bit vague and I do not want to add APIs that should not be there

Reply:
Good to me, but lets add an issue so that we do not forget :)